query-engine/driver-adapters/js/driver-adapter-utils/src/types.ts

```diff
@@ -96,6 +96,13 @@ export interface Transaction extends Queryable {
    * Rolls back the transaction.
    */
   rollback(): Promise<Result<void>>
+  /**
+   * Discards and closes the transaction which may or may not have been committed or rolled back.
+   * This operation must be synchronous. If the implementation requires calling creating new
```

Comment:
Not sure if requiring a synchronous function is the best approach, perhaps it would be better to let it be async, and either add a method to `AsyncJsFunction` that allows calling it in the background ignoring the result, or spawn a new tokio task in `JsTransaction::drop` that will actually await it and maybe log an error if it happens.

Reply:
It's not even about throughput, we'll have to do exactly the same amount of work synchronously until the first yield point regardless of whether this function is sync or async. The implementation of this method for Turso and PlanetScale is only "synchronous" in that it doesn't return a Promise and spawns a task on the event loop without waiting for it, and the implementation for pg and Neon would still only use synchronous operations even if the method was marked as async.

So changing this to be async should not really affect the performance or improve throughput but might arguably be a little cleaner, and might be a little more convenient when implementing new driver adapters (it will be harder to accidentally end up with unhandled promise rejections there).

Not something that we must decide right now, of course, we can improve this later.

Reply:
Thanks for the explanation. makes sense.